code/game/gamemodes/shadowling/special_shadowling_abilities.dm

```diff
@@ -33,6 +33,7 @@ var/list/possibleShadowlingNames = list("U'ruan", "Y`shej", "Nex", "Hel-uae", "N
 			for(var/obj/structure/alien/resin/wall/shadowling/R in shadowturf) //extremely hacky
 				qdel(R)
 				new /obj/structure/alien/weeds/node(shadowturf) //Dim lighting in the chrysalis -- removes itself afterwards
+			H.status_flags ^ GODMODE //Can't die while hatching
```

Comment:
What you should do is record the value of `status_flags` to a local variable, then set godmode using `|=`.  When you're done, set `status_flags` back to what it was originally.